examples/mpm_lagrangian_forces.py

```diff
@@ -175,13 +176,11 @@ def main():
         gui.circle((0.5, 0.5), radius=45, color=0x068587)
         # TODO: why is visualization so slow?
         particle_pos = x.to_numpy()
-        for i in range(n_elements):
-            for j in range(3):
-                a, b = vertices_[i, j], vertices_[i, (j + 1) % 3]
-                gui.line((particle_pos[a][0], particle_pos[a][1]),
-                         (particle_pos[b][0], particle_pos[b][1]),
-                         radius=1,
-                         color=0x4FB99F)
+        a = vertices_[:, :]
+        b = np.concatenate([vertices_[:, 1:], vertices_[:, 2:]], axis=1)
```

Comment:
Do i understand correctly that you're trying to do `np.roll(vertices_, shift=1, axis=1)`? ([`numpy.roll`](https://numpy.org/doc/stable/reference/generated/numpy.roll.html#numpy-roll)) Otherwise the last column of `a` and `b` will be identical?